### PR TITLE
Tweaking as_gt()

### PR DESCRIPTION
**Pull request by yihui**
I'm looking at the R scripts by alphabetical order, and reduced ~180 lines of code in `as_gt()`. Again, the changes may be easier to understand if you read the commits one by one.

In some places, I'm using my personal style (e.g., for line breaks and indentation), and happy to follow the style that other colleagues are comfortable with. In general, I tend to make the code a little compact so I can scroll less when reading the code.

I also realized `as_rtf()`'s source code is very similar to `as_gt()`, and perhaps can cut out more lines there lines after factoring out the common parts.

### CODE REVIEW

**Comment by jdblischak** [comment]
This is related to https://github.com/Merck/gsDesign2/issues/282

I still think that we eventually need to clean up the classes and methods as I noted in https://github.com/Merck/gsDesign2/issues/392. However, using `switch()` for now is definitely a great improvement in readability.

File: R/ahr_blinded.R
Comment (jdblischak) [inline R/ahr_blinded.R:120]: Personally I like the explicit return even though it is optional. And checking `git blame`, I'm pretty sure they existed before I started editing the code base. I'm happy to go along with the consensus though

Comment (yihui) [reply R/ahr_blinded.R:120]: I don't have a strong opinion here and can definitely use an explicit `return()`. Personally I use `return()` only when I need to return early.

File: R/ahr_blinded.R
Comment (LittleBeannie) [inline R/ahr_blinded.R:91]: Hi Yihui, I have a question regarding the error message output. Typically, our practice in gsDesign2 is to print the function name at the beginning of the error message. This consideration arises from the possibility that the `ahr_blinded` function could potentially be invoked within another function. So... I am wondering if we should keep the `ahr_blinded` here?

File: R/as_gt.R
Comment (LittleBeannie) [inline R/as_gt.R:285]: Echo with my previous comment, shall we add `as_gt` at the beginning of the error message?

**Review by LittleBeannie — COMMENTED** [review]
Thank you, @yihui, for tweaking the long `as_gt` function! I only have 2 minor comments, regarding the error message. Could you please take a look?

File: R/ahr_blinded.R
Comment (yihui) [reply R/ahr_blinded.R:91]: Good question. I deleted the function name in the error message because I thought it would be redundant.

```r
f = function() foo()
foo = function() {
  stop('No!')
}
f()
#> Error in foo() : No!
```

```r
foo = function() {
  stop('foo: No!')
}
f()
#> Error in foo() : foo: No!
```

This relates to a comment I made after https://github.com/Merck/gsDesign2/pull/448#issuecomment-2278528575 That is, the function name will be missing if you `do.call` the function.

```r
foo = function() {
  stop('No!')
}
f = function() do.call(foo, list())
f()
#> Error in (function ()  : No!
```

I think this case is relatively rare. If it does happen, `traceback()` can be helpful to know the chain of function calls.

That said, again I don't have a strong opinion here. In general, my principle is that I'll follow the preference of the project maintainer (i.e. you in this case).

File: R/ahr_blinded.R
Comment (yihui) [reply R/ahr_blinded.R:120]: @LittleBeannie Do you have a preference here? As I said, I'm totally fine with reverting to the original style `ans <- ...; return(ans)`, even though both the assignment and the explicit `return()` look unnecessary to me.

Comment (LittleBeannie) [reply R/ahr_blinded.R:120]: Thank you for verifying, @yihui ! I personally prefer the explicit return, as it provides a clearer message for non-advanced R programmers.

File: R/ahr_blinded.R
Comment (LittleBeannie) [reply R/ahr_blinded.R:91]: Thank you for the clear explanation; I understand your intention. I am open to this. If we choose to remove the function name at the beginning of the error message, we should review the entire gsDesign2 package, as the previous practice of gsDesign2 is to include the function name right at the start (maybe a new issue and PR).

Comment (yihui) [reply R/ahr_blinded.R:91]: Okay, as I'm going through the whole codebase, I'll do this along the way.

File: R/ahr_blinded.R
Comment (yihui) [reply R/ahr_blinded.R]: Got it. I've restored `return()` in 90352a04ad6d367b65777fd3915b5dc1ba5827fb.

**Review by LittleBeannie — APPROVED** [review]

File: R/ahr_blinded.R
Comment (jdblischak) [reply R/ahr_blinded.R:91]: I searched the repository for how `ahr_blinded()` is used. From the package code, examples, and tests, `ahr_blinded()` is always called directly. In other words, the function itself is never passed as an argument to another function. Thus the fact that an error comes from `ahr_blinded()` should be clear from the default error message, as @yihui noted above.

To demonstrate, I added `stop("there was an error")` to `ahr_blinded()` and generated the error message below:

```R
devtools::load_all()
ahr_blinded(
  surv = survival::Surv(
    time = simtrial::ex2_delayed_effect$month,
    event = simtrial::ex2_delayed_effect$evntd
  ),
  intervals = c(4, 100),
  hr = c(1, .55),
  ratio = 1
)
## Error in ahr_blinded(surv = survival::Surv(time = simtrial::ex2_delayed_effect$month,  : 
##   there was an error
```

File: R/as_gt.R
Comment (jdblischak) [inline R/as_gt.R:94]: This isn't equivalent to the `switch()` statement that was replaced, for example when `design_mtd <- "rd"`

```R
# before
[1] "Fixed Design of Risk Difference under Farrington-Manning Method"
# after
[1] "Fixed Design under Fixed Design of Risk Difference under Farrington-Manning Method"
```

**Review by jdblischak — CHANGES_REQUESTED** [review]
Overall this is looking great. Please see the one `switch()` statement that needs adjusted.

A bigger picture question for @LittleBeannie and @nanxstats : are we still planning on occasionally running {styler} on the code in {gsDesign2}? The default {styler} format is significantly different than Yihui's preferred style. Below is an example `git diff` after running {styler} on this PR:

image

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:94]: Good catch. Fixed now. Thanks!

**Review by jdblischak — APPROVED** [review]

**Pull request merged**
